pkgs/applications/editors/jetbrains/bin/update_bin.py

```diff
@@ -57,6 +59,19 @@ def download_sha256(url):
 channels = download_channels()
 
 
+def get_url(template, version_or_build_number, version_number):
+    parts = version_number.split(".")
+    for k in range(len(parts), 0, -1):
+        s = ".".join(parts[0:k])
+        url = template.format(version=version_or_build_number, versionMajorMinor=s)
+        try:
+            if urllib.request.urlopen(url).getcode() == 200:
+                return url
+        except HTTPError:
+            pass
+    return None
```

Comment:
I'm barely a python noob, so take all my feedback on the python with a very large grain on salt. 

Should we consider usign the Version type from `from packaging import version` (Line 12) to parse the versions rather than dealing with string manipulation ourselves? Especially for EAP packages which has slightly strange version numbers?

Reply:
Yes, that makes sense. I am a casual Python user, too, and I was unaware of the `version` module. Thanks.